kernel-spark/src/main/java/io/delta/kernel/spark/catalog/SparkTable.java

```diff
@@ -144,43 +140,11 @@ private SparkTable(
     this.snapshotManager = new PathBasedSnapshotManager(tablePath, hadoopConf);
     // Load the initial snapshot through the manager
     this.initialSnapshot = snapshotManager.loadLatestSnapshot();
-    this.schema = SchemaUtils.convertKernelSchemaToSparkSchema(initialSnapshot.getSchema());
-    this.partColNames =
-        Collections.unmodifiableList(new ArrayList<>(initialSnapshot.getPartitionColumnNames()));
 
-    final List<StructField> dataFields = new ArrayList<>();
-    final List<StructField> partitionFields = new ArrayList<>();
-
-    // Build a map for O(1) field lookups to improve performance
-    Map<String, StructField> fieldMap = new HashMap<>();
-    for (StructField field : schema.fields()) {
-      fieldMap.put(field.name(), field);
-    }
-
-    // IMPORTANT: Add partition fields in the exact order specified by partColNames
-    // This is crucial because the order in partColNames may differ from the order
-    // in snapshotSchema, and we need to preserve the partColNames order for
-    // proper partitioning behavior
-    for (String partColName : partColNames) {
-      StructField field = fieldMap.get(partColName);
-      if (field != null) {
-        partitionFields.add(field);
-      }
-    }
-
-    // Add remaining fields as data fields (non-partition columns)
-    // These are fields that exist in the schema but are not partition columns
-    for (StructField field : schema.fields()) {
-      if (!partColNames.contains(field.name())) {
-        dataFields.add(field);
-      }
-    }
-    this.dataSchema = new StructType(dataFields.toArray(new StructField[0]));
-    this.partitionSchema = new StructType(partitionFields.toArray(new StructField[0]));
-
-    this.columns = CatalogV2Util.structTypeToV2Columns(schema);
-    this.partitionTransforms =
-        partColNames.stream().map(Expressions::identity).toArray(Transform[]::new);
+    // Lazily compute schema-related metadata only when needed
+    // SparkSession.active() is deferred to when the schema is first accessed
+    this.schemaProvider =
+        new Lazy<>(() -> new SchemaProvider(SparkSession.active(), initialSnapshot));
```

Comment:
If we defer getting the schema until the API calls in `SchemaProvider`, we can avoid introducing `Lazy` here.